### DISCUSSION TITLE
Compile casks locally. No pre-build artifacts available.

### DISCUSSION THREAD

**Discussion by jkroepke** [Homebrew/discussions#3338] [Tap maintenance and Homebrew development]
Hi,

I'm currently want to offer a TAP which installs an application which is not develop my me. The application does not provide pre-build artifacts. Currently. I do not want to offer/publish or pre build artifacts.

Running `make build` inside the source code will produce an dmg file which can be moved to `/Applications/`. As a know, this pattern fits into concept of a cask. But I can not find an option to locally build a cask? OR should I write a formulae which builds the sources and moves the dmg file to  `/Applications/`?

**Comment by SMillerDev** [comment]
If it builds from source it's a formula. If it distributes a binary it's a cask.

**Reply by jkroepke** [reply]
But formula should install the application into the prefix? Since its an GUI application, it should be installed into `/Applications/`. Is this still fine?

**Reply by SMillerDev** [reply]
Formula will never install outside the prefix. But you can tell the users to symlink the app.
